### PR TITLE
Support Reddit as an OAuth provider

### PR DESCRIPTION
**Pull request by geuis**
Additionally enables basic http authentication header as required from reddit for oauth2 to function.

### CODE REVIEW

File: lib/providers/reddit.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// Load modules
+
+
+// Declare internals
+
+var internals = {};
+
+
+exports = module.exports = function (options) {
+    return {
```
Comment (geek) [inline lib/providers/reddit.js]: I need to add our lint rules... but we have an extra newline after every function declaration.

Please add a newline here and on line 17

Comment (geuis) [inline lib/providers/reddit.js]: updated

**Comment by geek** [comment]
I need to make it into an error, but here are the linting results from the build:

```
Linting results:
	lib/providers/reddit.js:
		Line 25: hapi/hapi-scope-start - Missing blank line at beginning of function.
	test/providers.js:
		Line 74: hapi/hapi-scope-start - Missing blank line at beginning of function.
		Line 85: hapi/hapi-scope-start - Missing blank line at beginning of function.
```

**Comment by geuis** [comment]
When I run eslint locally, no errors reported:

    eslint -c .eslintrc lib/providers/reddit.js 
    eslint -c .eslintrc test/providers.js

**Comment by geek** [comment]
@geuis thanks for all of the work on this, merging and publishing now :)

**Pull request merged**